src/mds/Server.cc

```diff
@@ -1154,7 +1170,7 @@ void Server::kill_session(Session *session, Context *on_safe)
        session->is_stale()) &&
       !session->is_importing()) {
     dout(10) << "kill_session " << session << dendl;
-    journal_close_session(session, Session::STATE_KILLING, on_safe);
+    journal_close_session(session, Session::STATE_KILLING, on_safe, mds->is_active());
```

Comment:
I think it's better to explicitly add a parameter to both  journal_close_session and kill_session.  bool meaning of ' on_active' is not clear

Reply:
updated